src/GitVersion.BuildAgents/Agents/AzurePipelines.cs

```diff
@@ -16,8 +16,24 @@ public override string[] GenerateSetParameterMessage(string name, string? value)
         $"##vso[task.setvariable variable=GitVersion.{name};isOutput=true]{value}"
     ];
 
-    public override string? GetCurrentBranch(bool usingDynamicRepos) => Environment.GetEnvironmentVariable("GIT_BRANCH")
-        ?? Environment.GetEnvironmentVariable("BUILD_SOURCEBRANCH");
+    public override string? GetCurrentBranch(bool usingDynamicRepos)
+    {
+        var gitBranch = Environment.GetEnvironmentVariable("GIT_BRANCH");
```

Comment:
Where does this variable come from? I can only find references that refer to [Jenkins](https://plugins.jenkins.io/git/#plugin-content-branch-variables)

Reply:
It's an environment variable we've adopted in GitVersion as a way for the build environment to indicate which branch is being built. But as you can see from #3049, support for the variable is not fully implemented.

Reply:
Any updates on this? It's annoying to have versions like `1.1.3-tags-1-1-1.1` created.